File: streaming/examples/twitter/twit_reduce.py
#!/usr/bin/env python

import sys
sys.path.append(".")

from pymongo_hadoop import BSONReducer

def reducer(key, values):
    print >> sys.stderr, "Processing Timezone %s" % key
    _count = 0
    for v in values:
        _count += v['count']
    return {'_id': key, 'count': _count}

BSONReducer(reducer)

File: streaming/examples/twitter/twit_reduce.rb
#!/usr/bin/env ruby
require 'mongo-hadoop'

# Function that takes key and array of values, iterates over all of the values,
# and returns a single document with the reduced data (summary) for that key.

MongoHadoop.reduce do |key, values|
  count = 0

  values.each do |value|
    count += value['count']
  end
  
  { :_id => key, :count => count }
end
